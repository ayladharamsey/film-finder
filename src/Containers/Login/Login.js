import React, { Component } from 'react';
import './Login.scss';
import { setUser} from '../../actions';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      buttonClick: false,
    }
  }

  updateUser = e => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleClick = e => {
    e.preventDefault()
    this.setState({buttonClick: true})
    this.props.loginUser({
      email: this.state.email,
      password: this.state.password
    })
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({
      email: '',
      password: '',
    });
  }

  handleError = () => {
    return (
      <h3>{this.props.hasErrored}</h3>
    )
  }

  linkToMovies = () => {
    if (this.props.user.email !== ""){
      return (
        <Redirect to='/movies'></Redirect>
      )
    }
  }

  render() {
    const { email, password, buttonClick } = this.state;
    const { hasErrored, user } = this.props;
    let error;

    if ((buttonClick === true)) {
      error = this.handleError()
    }

    if ((buttonClick === true) && (user.id !== undefined)) {
      error = this.linkToMovies()
    }

    return (
      <form className="form">
        <h1 className="title">FILMFINDER</h1>
          <section className="input_field">
            <Link to='/create-user'>
              <button className="sign_in-btn">create login</button>
            </Link>
            <h2 className="input_title" >User Login</h2>
            <input
              className="email"
              name="email"
              value={email}
              placeholder="Email"
              onChange={this.updateUser}
            />
            <input
              className="password"
              name="password"
              value={password}
              placeholder="Password"
              onChange={this.updateUser}
            />
            <button className="button" onClick={(e) => this.handleClick(e)}>
              login
            </button>
            {error}
          </section>
        </form>
    )
  }
}

export const mapStateToProps = state => ({
  hasErrored: state.hasErrored,
  user: state.user,
})

export const mapDispatchToProps = dispatch => ({
  setUser: user => dispatch(setUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Login);
