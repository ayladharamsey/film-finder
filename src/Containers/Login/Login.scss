input {
  color: black;
}

h3 {
  font-family: Source Sans Pro, sans-serif;
  color: #c71a1a;
  background-color: white;
  text-align: center;
  font-size: 13px;
  margin-top: 10px;
}

.button {
  width: 98%;
}