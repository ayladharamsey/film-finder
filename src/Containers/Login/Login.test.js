import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

describe('Login container tests', () => {

  it('Login should match its snapshot', () => {
    const wrapper = shallow( <Login /> )
    expect(wrapper).toMatchSnapshot();
  })
})