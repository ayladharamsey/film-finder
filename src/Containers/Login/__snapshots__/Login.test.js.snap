// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container tests Login should match its snapshot 1`] = `ShallowWrapper {}`;
